End of year retrospective 2024 (#342)
2i2c-org/2i2c-brochure: content/blog/2024/frx/index.md

---

> title: "`frx-challenge`: A new tool to host data challenges for Frictionless Research Exchanges"

Reviewer: ```suggestion
title: "`frx-challenges`: A new tool to host data challenges for Frictionless Research Exchanges"
```

date: 2024-12-06
---

2i2c is pleased to announce the `frx-challenges` project, a new open source tool to help communities host data challenges on shared infrastructure:

https://github.com/2i2c-org/frx-challenges

> This project aims to make it easier for communities to enable their users **submit code and data** that are **evaluated on secure infrastructure with access to private data and resources**.

Reviewer 2: > enable their users to 
(missing preposition I think)

{{< figure
  src="images/leaderboard.png"
  width="75%"

>   caption="An example leaderboard for a data challenge, taken from the [Cellmap Challenge](https://cellmapchallenge.janelia.org/). Users make submissions that are run against secure and private infrastructure and data, and provides feedback about the submission's performance. Learn more about the FRX challeenges project here: https://2i2c.org/frx-challenges/"

Reviewer 3: Small typo:

```suggestion
  caption="An example leaderboard for a data challenge, taken from the [Cellmap Challenge](https://cellmapchallenge.janelia.org/). Users make submissions that are run against secure and private infrastructure and data, and provides feedback about the submission's performance. Learn more about the FRX challenges project here: https://2i2c.org/frx-challenges/"
```

- **Evaluations must be totally secure**, so that submissions can't do nefarious things like mine cryptocurrency or gain access to the challenge's private data.
- **Evaluations must be automated**, so that running the challenge does not require extensive human intervention and can scale to many users.
- **Evaluation must be flexible**, so that the infrastructure can accept a variety of types of submissions (e.g. code, data, model weights, etc), run them with arbitrary environments designed by the organizers, and run them with the right hardware to get the job done.

These are just a few of the major challenges that we've tried to address with `frx-challenges`, and we're excited to see how it goes with our first assisted community challenge: the [Cellmap Challenge](https://cellmapchallenge.janelia.org/).

If you're interested in learning more or participating in this project, follow along at its GitHub repository:

https://github.com/2i2c-org/frx-challenges

This is still the **very early stages** of the project, and we imagine it will evolve significantly. We welcome feedback for how it can more effectively serve a variety of communities.

## Acknowledgements

> Many thanks to the Howard Hughes Medical Institute (HHMI) for collaborating with us on the [Cellmap Challenge](https://cellmapchallenge.janelia.org/), which led to the creation of this project.

Reviewer 2: this displays an extra space between the link and the comma that follows it
> [Cellmap Challenge](https://cellmapchallenge.janelia.org/)

Author: Oh nice this uncovered a bug in our link rendering in general that is now fixed!

categories: [impact]
---

2i2c is pleased to announce the `frx-challenges` project, a new open source tool to help communities host data challenges on shared infrastructure:

https://github.com/2i2c-org/frx-challenges

This project aims to make it easier for administrators to provide a service that enables users to **submit code and data** that are **evaluated on secure infrastructure with access to private data and resources**. It also provides a leaderboard that helps users compare their performance against others.

{{< figure
  src="images/leaderboard.png"
  width="75%"
  caption="An example leaderboard for a data challenge, taken from the [Cellmap Challenge](https://cellmapchallenge.janelia.org/). Users make submissions that are run against secure and private infrastructure and data, and provides feedback about the submission's performance. Learn more about the FRX challenges project here: https://2i2c.org/frx-challenges/"
>}}

It is designed to be lightweight and flexible, and can be run on a variety of shared infrastructure. For those who wish to run this project on cloud infrastructure, we've also published a Helm Chart to help you deploy `frx-challenges` with Kubernetes:

> https://2i2c.org/frx-challenges-helm-chart/

Reviewer 4: I would suggest not putting this so prominently, as it doesn't actually have any useful info right now on how to use it.

However there was a natural missing piece for **[FR-3 Challenges]**, and we could not identify any community-managed infrastructure that facilitated data challenges. This is the goal of `frx-challenges`.

## Why facilitate data challenges?

Data challenges are harder than you think! While it is simple enough to run somebody else's code locally, data challenges require a systematic, secure, and automated approach to accepting and evaluating submissions in a fair and repeatable way. Here are some of the big challenges to tackle:

- **Submissions must retain user and team identity**, which means that we must keep track of users and their submissions over time, since data challenges are designed to encourage iterative improvement and optimization.
- **Evaluations must use potentially complex resources and data** since many data challenges operate by publicly sharing a small dataset, and then running it against a much more complex dataset.

> - **Evaluations must be totally secure**, so that submissions can't do nefarious things like mine cryptocurrency or gain access to the challenge's private data.

Reviewer 4: ```suggestion
- **Evaluations must be totally secure**, so that submissions can't do nefarious things like mine cryptocurrency or exfiltrate the challenge's private data.
```

Reviewer 4: Since the submission must actually have access to the challenge's private data, but not actually be able to take the data itself back

- **Evaluations must be automated**, so that running the challenge does not require extensive human intervention and can scale to many users.
- **Evaluation must be flexible**, so that the infrastructure can accept a variety of types of submissions (e.g. code, data, model weights, etc), run them with arbitrary environments designed by the organizers, and run them with the right hardware to get the job done.

These are just a few of the major challenges that we've tried to address with `frx-challenges`, and we're excited to see how it goes with our first assisted community challenge: the [Cellmap Challenge](https://cellmapchallenge.janelia.org/).

If you're interested in learning more or participating in this project, follow along at its GitHub repository:

https://github.com/2i2c-org/frx-challenges

This is still the **very early stages** of the project, and we imagine it will evolve significantly. We welcome feedback for how it can more effectively serve a variety of communities.

## Acknowledgements

> Many thanks to the Howard Hughes Medical Institute (HHMI) for collaborating with us on the [Cellmap Challenge](https://cellmapchallenge.janelia.org/), which led to the creation of this project.

Reviewer 4: I'm not sure what org Kristen Ratan belongs to, but let's add them to the acknowledgement as well.
